Discogs.js

```
@@ -0,0 +1,119 @@
+	return "audioRecording";
+  }
+  return false;
+}
+
+function doWeb(doc, url) {
+  // Create a new Zotero item of type "audioRecording"
+  let item = new Zotero.Item("audioRecording");
+
+  // Locate the JSON-LD script tag
+  let jsonLdScript = doc.querySelector('script[type="application/ld+json"]');
+  if (!jsonLdScript) {
+	Zotero.debug("No JSON-LD found on the page.");
+	return;
+  }
+
+  try {
+	// Parse the JSON-LD content
+	let jsonLd = JSON.parse(jsonLdScript.textContent);
+
+	// Map JSON-LD fields to Zotero fields
+	if (jsonLd.name) {
+	  item.title = jsonLd.name; // e.g., "Electric / Impact (The Mixes)"
+	}
+
+	if (jsonLd.releaseOf && jsonLd.releaseOf.byArtist) {
+	  jsonLd.releaseOf.byArtist.forEach(function(artist) {
+		item.creators.push({
+		  lastName: artist.name, // e.g., "DJ Micro", "Vicious Vic"
+		  creatorType: "performer"
+		});
+	  });
+	}
+
+	if (jsonLd.recordLabel && jsonLd.recordLabel.length > 0) {
+	  item.label = jsonLd.recordLabel[0].name; // e.g., "Caffeine Records"
+	}
+
+	if (jsonLd.musicReleaseFormat) {
+	  item.medium = jsonLd.musicReleaseFormat; // e.g., "Vinyl"
+	}
+
+	if (jsonLd.datePublished) {
+	  item.date = jsonLd.datePublished; // e.g., 1996
+	}
+
+	if (jsonLd.genre) {
+	  if (Array.isArray(jsonLd.genre)) {
+		jsonLd.genre.forEach(genre => item.tags.push(genre));
+	  } else {
+		item.tags.push(jsonLd.genre); // e.g., "Electronic"
+	  }
+	}
+
+	if (jsonLd.catalogNumber) {
+	  item.callNumber = jsonLd.catalogNumber; // e.g., "CF 2833, L-46099"
+	}
+
+	if (jsonLd.description) {
+	  item.abstractNote = jsonLd.description; // e.g., "This is the black vinyl release..."
+	}
+
+	if (jsonLd.image) {
+	  item.attachments.push({
+		url: jsonLd.image, // e.g., cover image URL
+		title: "Cover Image",
+		mimeType: "image/jpeg"
+	  });
+	}
+
+	// Add the release page URL
+	item.url = url;
+
+	// Complete the item and save it to Zotero
+	item.complete();
+  } catch (e) {
+	Zotero.debug("Error parsing JSON-LD: " + e.message);
```

Don't wrap in a try-catch - just let it fail.

---

Just item.complete(); then ?

---

Not sure what you mean - just replace the entire try-catch with the `try` block's contents.